src/weapon.py

```diff
@@ -101,6 +97,8 @@ class Rocket(Weapon):
     def setup(self):
         self.rocket_speed = 300
         self.fire_rate = 0.5
+        #the -45 degree angle in the offset corrects for rocket texture angled up 45 degrees
+        self.muzzle_offset = (30, 2, -45)
```

Comment:
A good comment to clarify an odd value here.  Since this is a change to handle the sprite texture, can this be moved to the SpriteForRocket class?  If this is a generalized problem, we probably need a standard way of handling this kind of texture rotation in the future

Reply:
Moving into `SpriteForRocket` makes sense.  Putting rotation on `muzzle_offset` implies that the muzzle is facing 45 degrees off of center, but it's not.

I think the generalized "transformation" (currently named "offset") functions we have are a good way to handle this.  The sprite is positioned at some transformation relative to its owner, which itself has a transformation.

Reply:
I think this angle offset will eventually belong in the Projectile class once this composition is built. The `SpriteForRocket` class which will probably get generalized to `SpriteForProjectile` is just so the sprite can call the projectile that owns it, so I don't think this subclass should have other parameters